mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -21,7 +21,7 @@
 from inspect import signature
 from itertools import chain
 from os.path import join, abspath, dirname, basename, exists, isfile
-from threading import Event
+from threading import Event, Lock
```

Comment:
I've started using `RLock` for applications like these in case different wrappers are added/refactored; so long as only one thread is mutating things at a time, we should be okay